atef/widgets/config.py

```diff
@@ -1765,10 +1803,319 @@ def __init_subclass__(cls, *args, **kwargs):
         CompView.register_comparison(cls.data_type, cls)
 
 
-# This class should be replaced by a real "Equals" widget
-class EqualsWidget(CompMixin, QLabel):
+def user_string_to_bool(text: str) -> bool:
+    """
+    Interpret a user's input as a boolean value.
+
+    Strings like "true" should evaluate to True, strings
+    like "fa" should evaluate to False, numeric inputs like
+    1 or 2 should evaluate to True, numeric inputs like 0 or
+    0.0 should evaluate to False, etc.
+
+    Parameters
+    ----------
+    text : str
+        The user's text input as a string. This is usually
+        the value directly from a line edit widget.
+    """
+    if not text:
+        return False
+    try:
+        if text[0].lower() in ('n', 'f', '0'):
+            return False
+    except (IndexError, AttributeError):
+        # Not a string, let's be slightly helpful
+        return bool(text)
+    return True
+
+
+class EqualsWidget(CompMixin, AtefCfgDisplay, QWidget):
+    """
+    Widget to handle the fields unique to the "Equals" Comparison.
+
+    Parameters
+    ----------
+    bridge : QDataclassBridge
+        Dataclass bridge to an "Equals" object. This widget will
+        read from and write to the "value", "atol", and "rtol"
+        fields.
+    parent : QObject, keyword=only
+        The normal qt parent argument
+    """
+    filename = 'comp_equals.ui'
     data_type = Equals
+    label_to_type: Dict[str, type] = {
+        'float': float,
```

Comment:
Hmm, could we instead create an appropriate widget based on the selected data type?
bool -> checkbox, str -> line edit, int/float -> spinbox/double spinbox

Reply:
You're right, I should have considered widget types here. Checkbox is absolutely a better interface for the booleans, spinbox is a marginal improvement for the numeric types.

Reply:
Actually I'd almost prefer a true/false combobox over a checkbox for the booleans

Reply:
I think that is actually clearer - agreed!